waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingServiceTest.java

```diff
@@ -355,4 +360,127 @@ public void panopticOperationsHandlerSetUgi() throws Exception {
     assertThat(result, containsInAnyOrder("ugi", "ugi2"));
   }
 
+  @Test(expected = NoPrimaryMetastoreException.class)
+  public void noPrimaryMappingThrowsException() {
+    when(metaStoreMappingFactory.newInstance(federatedMetastore)).thenReturn(metaStoreMappingFederated);
+    AbstractMetaStore unavailableMetastore = newFederatedInstance("name2", "thrift:host:port");
+    MetaStoreMapping unavailableMapping = mockNewMapping(false, "name2_");
+    when(metaStoreMappingFactory.newInstance(unavailableMetastore)).thenReturn(unavailableMapping);
+
+    service = new PrefixBasedDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(federatedMetastore, unavailableMetastore), queryMapping);
+    service.primaryDatabaseMapping();
+  }
+
+  @Test(expected = NoPrimaryMetastoreException.class)
+  public void noPrimaryThrowsExceptionForUnmappedDatabase() {
+    when(metaStoreMappingFactory.newInstance(federatedMetastore)).thenReturn(metaStoreMappingFederated);
+    AbstractMetaStore unavailableMetastore = newFederatedInstance("name2", "thrift:host:port");
+    MetaStoreMapping unavailableMapping = mockNewMapping(false, "name2_");
+    when(metaStoreMappingFactory.newInstance(unavailableMetastore)).thenReturn(unavailableMapping);
+
+    service = new PrefixBasedDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(federatedMetastore, unavailableMetastore), queryMapping);
+    String databaseName = "some_unknown_prefix_db";
+    try {
+      service.databaseMapping("some_unknown_prefix_db");
+    } catch (NoPrimaryMetastoreException noPrimaryException) {
+      String expectedErrorMessage = "Waggle Dance error no database mapping available tried to map database '"
```

Comment:
It can be quite brittle to assert exception messages as changing the message will break this test and we probably don't care _that_ much about the actual message in this case. I'll leave it up to you whether you want to keep this here or just remove the assertion and the whole try/catch/